tests/test_security_gssapi.cpp

```diff
@@ -179,19 +185,23 @@ void test_unauth_creds (void *ctx, void *server, void *server_mon, char *endpoin
     rc = zmq_setsockopt (client, ZMQ_GSSAPI_PRINCIPAL,
                          name, strlen (name) + 1);
     assert (rc == 0);
+#ifdef ZMQ_BUILD_DRAFT_API
     int name_type = ZMQ_GSSAPI_NT_HOSTBASED;
     rc = zmq_setsockopt (client, ZMQ_GSSAPI_PRINCIPAL_NAMETYPE,
                          &name_type, sizeof (name_type));
     assert (rc == 0);
+#endif
     zap_deny_all = 1;
     rc = zmq_connect (client, endpoint);
     assert (rc == 0);
 
     expect_bounce_fail (server, client);
     close_zero_linger (client);
 
+#ifdef ZMQ_BUILD_DRAFT_API
     int event = get_monitor_event (server_mon, NULL, NULL);
-    assert (event == ZMQ_EVENT_HANDSHAKE_FAILED);
+    assert (event == ZMQ_EVENT_HANDSHAKE_FAILED_AUTH);
```

Comment:
@sigiesec I took a very wild guess here and in the next events, is this correct?